### DISCUSSION TITLE
new-tpp certificate keyusage & custom fields?

### DISCUSSION THREAD

**Discussion by fearnie85** [Venafi/venafips#23] [Ideas]
Hi. 

Is it possible to set key usage with new-tppcertificate? We use a template cert as our CA and add this as a root. This sets most things, but occasion, we need others to be set.
If not, I can make the cert using powershell and then import it to venafi. But how can I set CertificateAuthorityDN '\ved\policy\CA Templates\my template' When using venafips to import the certs? I really like this feature. 

Also. with New-Tppcertificate, how do I set custom fields? do you have an example script I can read please.

Thanks.

**Comment by wilddev65** [comment]
Key Usage in general is not something that is able to be set in Venafi. As such there isn't an API parameter for it. I would ask why you need to do this, as in general in TPP, key usage isn't something that any end user is concerned about having set, most don't even know what it is. What is your use case for it and why are you needing to set it?
If it's something that an end user of yours requires to be set in their certificate, then they would need to generate the CSR themselves and then import that into TPP for signing.
As to setting a custom field, you can do this with Set-TPPAttribute. It's not currently available in New-TPPCertificate.
